### PR TITLE
More efficient header use

### PR DESCRIPTION
**Pull request by benaadams**
Raised issue https://github.com/dotnet/runtime/issues/52507 for `operator+` on `StringValues` which would make some of the changes prettier

### CODE REVIEW

File: src/Middleware/Localization/src/RequestLocalizationMiddleware.cs [hunk 1]
```diff
@@ -126,7 +126,10 @@ public async Task Invoke(HttpContext context)
 
             if (_options.ApplyCurrentCultureToResponseHeaders)
             {
-                context.Response.Headers.Add(HeaderNames.ContentLanguage, requestCulture.UICulture.Name);
+                var headers = context.Response.Headers;
+                // Can't use += as StringValues does not override operator+
+                // and the implict conversions will cause an incorrect string concat https://github.com/dotnet/runtime/issues/52507
+                headers.ContentLanguage = StringValues.Concat(headers.ContentLanguage, requestCulture.UICulture.Name);
```
Comment (BrennanConroy) [inline src/Middleware/Localization/src/RequestLocalizationMiddleware.cs:132]: This is a behavior change, `Add` throws for duplicates, I assume the change is fine but want to call it out.
+@Tratcher for your opinion

**Review by BrennanConroy — APPROVED** [review]
Looks good, just want to make sure the Add -> Append is ok

File: src/Middleware/Localization/src/RequestLocalizationMiddleware.cs (hunk 1 above)
Comment (Tratcher) [reply src/Middleware/Localization/src/RequestLocalizationMiddleware.cs:132]: Most uses of Add are a mistake anyways, headers should always be Set or Append'd. Let's change this to Set to keep it consistent.

Comment (Tratcher) [inline src/Middleware/Localization/src/RequestLocalizationMiddleware.cs:132]:
```suggestion
                headers.ContentLanguage = requestCulture.UICulture.Name;
```

**Review by Tratcher — APPROVED** [review]

**Pull request merged**